codec-redis/src/main/java/io/netty/handler/codec/redis/RedisEncoder.java

```diff
@@ -178,20 +186,27 @@ private void writeArrayHeader(ByteBufAllocator allocator, boolean isNull, long l
         if (isNull) {
             final ByteBuf buf = allocator.ioBuffer(RedisConstants.TYPE_LENGTH + RedisConstants.NULL_LENGTH +
                                                    RedisConstants.EOL_LENGTH);
-            buf.writeByte(RedisMessageType.ARRAY_HEADER.value());
+            writeMessageType(buf, RedisMessageType.ARRAY_HEADER);
             buf.writeShort(RedisConstants.NULL_SHORT);
             buf.writeShort(RedisConstants.EOL_SHORT);
             out.add(buf);
         } else {
             final ByteBuf buf = allocator.ioBuffer(RedisConstants.TYPE_LENGTH + RedisConstants.LONG_MAX_LENGTH +
                                                    RedisConstants.EOL_LENGTH);
-            buf.writeByte(RedisMessageType.ARRAY_HEADER.value());
+            writeMessageType(buf, RedisMessageType.ARRAY_HEADER);
             buf.writeBytes(numberToBytes(length));
             buf.writeShort(RedisConstants.EOL_SHORT);
             out.add(buf);
         }
     }
 
+    private static void writeMessageType(ByteBuf buf, RedisMessageType type) {
+        if (type.length() == 0) {
```

Comment:
instead of `length` here, we could just let `RedisMessageType` write itself to the `buf`:

```java
enum RedisMessageType {
  public void serialize(ByteBuf but) {
    if (value != null) { buf.writeByte(value); }
  }
}
```